app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -287,6 +289,13 @@ class GleanMetricsService(private val context: Context) : MetricsService {
             }
 
             activationPing.checkAndSend()
+
+            // We need to ensure this function is called after Glean is initialized because the service-experiments
+            // component requires Glean to be initialized first in case it needs to record metrics.
+            Experiments.initialize(
```

Comment:
Please note that `GleanMetricsService.start()` is being called every single time the telemetry option is toggled on/off in the UI.  After [this part](https://github.com/mozilla-mobile/fenix/blob/390659fe58fa99064b051983f39d183636a8e397/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L264) it only gets called once. Is there any mechanism to disable experiments when telemetry is turned off? Is this needed? Does it need to be hooked up to a separate pref? That's probably a question for @mythmon (maybe we want a similar behaviour as Desktop?) and for Fenix people :)

Reply:
On Desktop, Normandy (which includes an experiments subsystem, but also other things like gradual rollouts) currently disables itself entirely when Telemetry is disabled. This has proven to be a problem for a few reasons.

The first is that there are many cases, where we might want the experiments program to be active without telemetry, such as QA. For example, mozregression builds have Telemetry disabled, and so Normandy is difficult to run regressions on.

The second is because Normandy does more than just experiments, there are plenty of times where we want to deliver something to users even if they don't have Telemetry enabled (Armagaddon, for a recent example).

The second doesn't apply to this integration, but the first does. I'd suggest making it independent of Telemetry. For a settings UI it might make sense to tie the two preferences together into one toggle, since I imagine that's what most users want.

Reply:
Totally agree with keeping this separate from the telemetry preference based on that logic.  Do we want to have a completely separate preference here or treat it as always on like fretboard currently is treated in Fenix?

Reply:
I know that there are users that are ok keeping Telemetry on, but don't want experiments running. They are happy to contribute technical data about the product, but don't want it changing in unexpected ways. I think offering a toggle, even if it is in some "advanced" panel, would be a nice thing to offer to users.

I also don't really know how this affects the implementation. I'm still getting my bearings. If this is really hard then I think the fretboard approach would be a fine compromise.

Reply:
This is the first time I've had to touch the Fenix codebase, so it's hard to say what the effort of adding a new pref/toggle will be but I'm looking into it before I decide which way to go.

Reply:
@travis79 this is a product specific question, so I'd recommend to also bring in this discuss a representative from the Fenix team.

> The first is that there are many cases, where we might want the experiments program to be active without telemetry, such as QA. For example, mozregression builds have Telemetry disabled, and so Normandy is difficult to run regressions on.

Does this apply here, though? When performing validation tasks, recently, QA used released binaries or nightly builds (both of them having telemetry on). I'm not sure there's a case in which one would have telemetry off outside of local testing builds.

Reply:
I don't guess it had dawned on me until just now that this may be a moot point about having the init behind it's own preference because currently, Experiments requires Glean to be initialized first or it just won't init itself, see [here](https://github.com/mozilla-mobile/android-components/blob/663e67fc17bf3747afb76f202dc05627501cc2a3/components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt#L65).

I'm not sure we can maintain a hard dependency on Glean like that if we plan on allowing the two libraries to be used separately from each other.

Reply:
This may also benefit from the changes to recording metrics prior to initialization that are being proposed for the crash counts [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1566917) and may make the initialization order a little easier to handle.